### PR TITLE
Update the migration guide

### PR DESCRIPTION
**Pull request by joaander**
## Description

Briefly describe some of the C++ API changes in the migration guide. Also mention some more removed functionalities.

## Motivation and context

Users have been asking about these changes on the mailing list. This change formally documents the changes. Going forward, we should try to update the migration guide with any relevant notes in every PR.

## How has this been tested?

I reviewed the compiled sphinx docs locally.

## Change log

```
- Update migration guide
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: sphinx-doc/migrating.rst [hunk 1]
```diff
@@ -101,3 +101,21 @@ Components
   - Replace all ``target_link_libraries`` and ``set_target_properties`` with
     ``target_link_libraries(_${COMPONENT_NAME} PUBLIC HOOMD::_hoomd)`` (can link
     ``HOOMD::_md``, ``HOOMD::_hpmc``, etc. if necessary).
+
+* Numerous C++ class APIs have changed, been removed, or renamed. Review the
+  header files to see new class signatures. These changes may require you to
+  update your compoent accordingly. Some of the more notable changes include:
```
Comment (SchoeniPhlippsn) [inline sphinx-doc/migrating.rst:107]:
```suggestion
  update your component accordingly. Some of the more notable changes include:
```

File: sphinx-doc/migrating.rst [hunk 2]
```diff
@@ -101,3 +101,21 @@ Components
...
+
+  - ``Variant`` has been completely rewritten.
+  - ``Trigger`` replaces periodic and variable perioud scheduling.
+  - ``NeighborList`` has a ``addRCutMatrix`` method clients must use to specify
+    the maximum cutoff radii per type pair.
+  - ``timestep`` is now of type ``uint64_t``.
+  - ``RandomGenerator`` is now constructed with a ``Seed`` and ``Counter``
+    object that support 64-bit timesteps.
+  - ``m_seed`` is no longer present in indididual operation objects. Use the
```
Comment (SchoeniPhlippsn) [inline sphinx-doc/migrating.rst:116]:
```suggestion
  - ``m_seed`` is no longer present in individual operation objects. Use the
```

File: sphinx-doc/migrating.rst [hunk 3]
```diff
@@ -101,3 +101,21 @@ Components
...
+    global seed provided by ``SystemDefinition``.
+  - The HPMC integrators have been heavily refactored.
+  - ``ParticleGroup`` instances are now constructed from immutable, reusable,
+    and user-customizable ``ParticleFilter`` instances.
+  - All GPU code is now written with HIP to support NVIDIA and AMD GPUs.
```
Comment (SchoeniPhlippsn) [inline sphinx-doc/migrating.rst:121]: One more thing that has been changed is the Active forces. The active force direction is now coupled to the orientation of the particles and set in the particle reference frame.

File: sphinx-doc/deprecated.rst [hunk 4]
```diff
@@ -77,6 +77,10 @@ Commands and features deprecated in v2.x will be removed in v3.0.
      - Replace with
    * - ``group`` parameter to ``integrate.mode_minimize_fire``
      - Pass group to integration method.
+   * - ``alpha`` parameter to ``pair.lj`` and related classes
+     - n/a
+   * - LJ 12-8 pair potential
+     - Mie potential
```
Comment (SchoeniPhlippsn) [inline sphinx-doc/deprecated.rst:83]: Here you can also mention the active forces: the array of tuples f_list and t_list have been replaced with the tuples active_force and active_torque

**Review by SchoeniPhlippsn — COMMENTED** [review]
Only found a couple of typos and mentioned the new active force API, which should be mentioned as well.

File: sphinx-doc/migrating.rst [hunk 5]
```diff
@@ -101,3 +101,21 @@ Components
   - Replace all ``target_link_libraries`` and ``set_target_properties`` with
     ``target_link_libraries(_${COMPONENT_NAME} PUBLIC HOOMD::_hoomd)`` (can link
     ``HOOMD::_md``, ``HOOMD::_hpmc``, etc. if necessary).
+
+* Numerous C++ class APIs have changed, been removed, or renamed. Review the
+  header files to see new class signatures. These changes may require you to
+  update your compoent accordingly. Some of the more notable changes include:
+
+  - ``Variant`` has been completely rewritten.
+  - ``Trigger`` replaces periodic and variable perioud scheduling.
```
Comment (b-butler) [inline sphinx-doc/migrating.rst:110]:
```suggestion
  - ``Trigger`` replaces periodic and variable period scheduling.
```

**Review by b-butler — COMMENTED** [review]
Found one typo, otherwise LGTM.

File: sphinx-doc/deprecated.rst (hunk 4 above)
Comment (joaander) [reply sphinx-doc/deprecated.rst:83]: done

File: sphinx-doc/migrating.rst (hunk 3 above)
Comment (joaander) [reply sphinx-doc/migrating.rst:122]: done

**Review by b-butler — APPROVED** [review]
LGTM

**Pull request merged**